cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -321,6 +322,17 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 
 	// IsEmptyStatus is to check if initial status of cVR object is empty.
 	if IsEmptyStatus(cVR) || IsInitStatus(cVR) || IsRecreateStatus(cVR) {
+		// We should generate replicaID for new volume only
+		if IsEmptyStatus(cVR) || IsInitStatus(cVR) {
+			if err := volumereplica.GenerateReplicaID(cVR); err != nil {
+				klog.Errorf("cVR ReplicaID creation failure: %v", err.Error())
+				return string(apis.CVRStatusOffline), err
+			}
+		}
+		if len(cVR.Spec.ReplicaID) == 0 {
+			return string(apis.CVRStatusOffline), merrors.New("ReplicaID is not set")
+		}
+
```

Comment:
Let us assume that `zfs dataset` is creation is successful and from here we are returning status since cvr is a pointer it will have replicaId now if update call fails then we are not populating replicaId which will be a problem.
In the previous commit, it was there but now.
cc: @vishnuitta @mynktl

Reply:
good one @mittachaitu .. so, reconcilation for this attribute is not happening.. can you please add this to hacktoberfest link?